src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -186,108 +205,165 @@ internal void UpdateSelectedItem(PackageItemListViewModel selectedItem)
+                await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+
+                var errorMessage = ExceptionUtilities.DisplayMessage(ex);
+                _logger.Log(ProjectManagement.MessageLevel.Error, errorMessage);
+
+                _loadingStatusIndicator.SetError(errorMessage);
+
+                _loadingStatusBar.SetError();
+                _loadingStatusBar.Visibility = Visibility.Visible;
+            }
+            finally
+            {
+                if (_loadingStatusIndicator.Status != LoadingStatus.NoItemsFound
+                    && _loadingStatusIndicator.Status != LoadingStatus.ErrorOccurred)
                 {
-                    loadCts.Cancel();
-                    loadCts.Dispose();
-                    currentLoader.Reset();
+                    // Ideally, After a search, it should report its status and,
+                    // do not keep the LoadingStatus.Loading forever.
+                    // This is a workaround.
+                    var emptyListCount = addedLoadingIndicator ? 1 : 0;
+                    if (Items.Count == emptyListCount)
+                    {
+                        _loadingStatusIndicator.Status = LoadingStatus.NoItemsFound;
+                    }
+                    else
+                    {
+                        Items.Remove(_loadingStatusIndicator);
+                    }
+                }
+            }
 
-                    // Write stack to activity log
-                    Mvs.ActivityLog.LogError(LogEntrySource, ex.ToString());
+            UpdateCheckBoxStatus();
 
-                    await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+            LoadItemsCompleted?.Invoke(this, EventArgs.Empty);
+        }
 
-                    var errorMessage = ExceptionUtilities.DisplayMessage(ex);
-                    _logger.Log(ProjectManagement.MessageLevel.Error, errorMessage);
+        internal void FilterItems(ItemFilter itemFilter, CancellationToken token)
+        {
+            if (!Items.Contains(_loadingStatusIndicator))
+            {
+                Items.Add(_loadingStatusIndicator);
+            }
+            _loadingStatusIndicator.Status = LoadingStatus.Loading;
 
-                    _loadingStatusIndicator.SetError(errorMessage);
+            // If there is another async loading process - cancel it.
+            var loadCts = CancellationTokenSource.CreateLinkedTokenSource(token);
+            Interlocked.Exchange(ref _loadCts, loadCts)?.Cancel();
 
-                    _loadingStatusBar.SetError();
-                    _loadingStatusBar.Visibility = Visibility.Visible;
+            try
+            {
+                if (itemFilter == ItemFilter.UpdatesAvailable)
+                {
+                    ApplyItemsFilterForUpdatesAvailable();
+                }
+                else
+                {
+                    //Show all the items, without an Update filter.
+                    ClearItemsFilter();
```

Comment:
the term items filter is a bit confusing...mostly because itemFilter is really a variable that is closer to activeTab.
so when you say ClearItemsFilter(); ... you really mean ... ShowAllInstalledPackages() right?

basically this feedback is that we shouldn't use our two different filter concepts with the same terminology.

Reply:
I'd rather refactor itemFilter to something else  (as you say, active tab) rather than call this something else. 
I'm going to be applying more filtering to more items in the future, so no, I don't want to make this ShowAllInstalledPackages().

Reply:
Changed it to ClearUIFilter since that's what it's doing. It doesn't matter if it's AllInstalled or not, because this is removing *any* UI filtering.